File: src/service/AttendanceService.java
package service;

import data.AttendanceDAO;

/**
 * Service Layer - Business Logic
 */
public class AttendanceService {

    private AttendanceDAO attendanceDAO;

    // Constructor Injection
    public AttendanceService(AttendanceDAO attendanceDAO) {
        this.attendanceDAO = attendanceDAO;
    }

    public void markAttendance(String studentId, String date, String status) {
        // Business rules can be added here
        attendanceDAO.saveAttendance(studentId, date, status);
    }

    public void viewAttendance(String studentId) {
        attendanceDAO.getAttendanceByStudent(studentId);
    }
}

File: src/service/AttendanceService.py
# service/attendance_service.py
from config.dao_factory import DAOFactory

class AttendanceService:
    def __init__(self):
        self.dao = DAOFactory.get_attendance_dao()
        self.strategy = None

    def set_strategy(self, strategy):
        self.strategy = strategy

    def mark_attendance(self, student):
        if self.strategy and self.strategy.mark_attendance(student):
            self.dao.save_attendance(student.id, "Present")
